### PR TITLE
reworks builder pattern and adds interface for published pipelines

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: archive/src/main/java/org/opentestsystem/rdw/archive/S3ArchiveService.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public S3ArchiveService(final ArchiveProperties archiveProperties) {
         this.amazonS3 = AmazonS3ClientBuilder.standard()
                 .withRegion(archiveProperties.getS3RegionStatic())
                 .withCredentials(
-                        new StaticCredentialsProvider(
```
Comment (esotrope) [inline archive/src/main/java/org/opentestsystem/rdw/archive/S3ArchiveService.java]: this was deprecated

File: script/src/main/java/org/opentestsystem/rdw/script/PipelineDefinition.java [hunk 2]
```diff
@@ -33,61 +25,68 @@ public String getRollbackVersion() {
...
+    public String getPublishedBy() {
+        return publishedBy;
+    }
 
-        public PipelineDefinitionBuilder name(final String name) {
-            this.name = name;
-            return this;
-        }
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
```
Comment (esotrope) [inline script/src/main/java/org/opentestsystem/rdw/script/PipelineDefinition.java]: less hands-on way to string build

File: script/src/main/java/org/opentestsystem/rdw/script/ScriptDefinition.java [hunk 3]
```diff
@@ -1,59 +1,51 @@
...
-     * @param input an open Reader with the input file for the script to work on
+     * @param input                   an open Reader with the input file for the script to work on
      * @param additionalConfiguration a Map of string properties that will be added to the configuration and
      *                                made available to the scripts
      * @return an instance of a PipelineScript
-     *
      * @throws Exception if there is an problem constructing the script
      */
-    public PipelineScript getInstance(Object input, Map<String,Object> additionalConfiguration) throws Exception {
+    public PipelineScript getInstance(Object input, Map<String, Object> additionalConfiguration) throws Exception {
```
Comment (esotrope) [inline script/src/main/java/org/opentestsystem/rdw/script/ScriptDefinition.java]: if it were me i'd keep the model dumb and put these get instance methods somewhere expected to have brains like a service

File: script/src/main/java/org/opentestsystem/rdw/script/PipelineDefinitionRepository.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+public interface PipelineDefinitionRepository {
+
+    /**
+     * Finds all of the published pipelines matching the given pipeline code and version
+     *
+     * @param code The pipeline code to filter on
+     * @param version The pipeline version to filter on if any
+     * @return All published pipelines for the given pipeline code
+     */
+    List<PipelineDefinition> findAllByCodeAndVersion(String code, Integer version);
```
Comment (malaffoon) [inline script/src/main/java/org/opentestsystem/rdw/script/PipelineDefinitionRepository.java]: Uhmm, shouldn't there be only one instance per code and version?

Comment (esotrope) [reply script/src/main/java/org/opentestsystem/rdw/script/PipelineDefinitionRepository.java]: you're correct. this was in avoidance of exposing two repo methods to service one controller method. The endpoint is `/publishedPipelines?pipelineCode=X&version=Y`

Comment (malaffoon) [reply script/src/main/java/org/opentestsystem/rdw/script/PipelineDefinitionRepository.java]: Ok. I guess my point is: should it be returning a collection or just a single thing?

Comment (esotrope) [reply script/src/main/java/org/opentestsystem/rdw/script/PipelineDefinitionRepository.java]: it says a collection but it's not - i do `(List) singleThing` : J
totally kidding and i'll add another method so it's not confusing

**Pull request merged**